nav2_costmap_2d/src/footprint.cpp

```diff
@@ -203,53 +203,38 @@ bool makeFootprintFromString(const std::string & footprint_string,
   return true;
 }
 
-
-std::vector<geometry_msgs::msg::Point> makeFootprintFromParams(rclcpp::Node::SharedPtr nh)
+std::vector<geometry_msgs::msg::Point> makeFootprintFromParams(rclcpp::Node::SharedPtr node)
 {
-  auto parameters_client = std::make_shared<rclcpp::SyncParametersClient>(nh);
-
   std::string full_param_name = "footprint";
   std::string full_radius_param_name = "robot_radius";
   std::vector<geometry_msgs::msg::Point> points;
+  std::string footprint;
 
-  if (parameters_client->has_parameter(full_param_name)) {
-    XmlRpc::XmlRpcValue footprint_xmlrpc;
-    footprint_xmlrpc = parameters_client->get_parameter<XmlRpc::XmlRpcValue>(full_param_name);
-    if (footprint_xmlrpc.getType() == XmlRpc::XmlRpcValue::TypeString &&
-        footprint_xmlrpc != "" && footprint_xmlrpc != "[]")
-    {
-      if (makeFootprintFromString(std::string(footprint_xmlrpc), points)) {
-        writeFootprintToParam(nh, points);
-        return points;
-      }
-    } else if (footprint_xmlrpc.getType() == XmlRpc::XmlRpcValue::TypeArray) {
-      points = makeFootprintFromXMLRPC(footprint_xmlrpc, full_param_name);
-      writeFootprintToParam(nh, points);
+  node->get_parameter_or<std::string>(full_param_name, footprint, "[]" );
+  if (footprint != "" && footprint != "[]")
+  {
+    if (makeFootprintFromString(std::string(footprint), points)) {
+      writeFootprintToParam(node, points);
       return points;
     }
   }
 
-  if (parameters_client->has_parameter(full_radius_param_name)) {
-    double robot_radius;
-    robot_radius = parameters_client->get_parameter<double>(full_radius_param_name, 1.234);
-    points = makeFootprintFromRadius(robot_radius);
+  double robot_radius;
+  node->get_parameter_or<double>(full_radius_param_name, robot_radius, 0.46);
```

Comment:
I'd set this radius to 0.1 as well. Actually, it shouldn't be necessary since you used `set_if_not_set` in the constructor. This should always be set now.